enigma-principal/app/src/controller/boot_network.rs

```diff
@@ -0,0 +1,123 @@
+use std::sync::Arc;
+use std::{fs::File, io::prelude::*, path::Path};
+
+use failure::Error;
+use sgx_types::sgx_enclave_id_t;
+use structopt::StructOpt;
+use rustc_hex::{FromHex, ToHex};
+use crossbeam_utils::thread;
+
+use controller::{
+    km_http_server::KMHttpServer,
+    km_utils::{
+        KMConfig,
+        SgxEthereumSigner,
+        PrivateKeyEthereumSigner
+    },
+};
+use common_u::custom_errors::{ControllerError, EnclaveError, VerifierError, ConfigError};
+use esgx::general::ENCLAVE_DIR;
+use esgx::equote;
+use controller;
+use enigma_crypto::EcdsaSign;
+use enigma_tools_u::{esgx::general::storage_dir, web3_utils::enigma_contract::EnigmaContract};
+use controller::km_controller::KMController;
+use epochs::epoch_trigger::EthereumTracker;
+
+pub fn create_signer(eid: sgx_enclave_id_t, with_private_key: bool, private_key: &[u8]) -> Box<dyn EcdsaSign + Send + Sync> {
+    if with_private_key {
+        let mut pk_32 = [0u8; 32];
+        pk_32.copy_from_slice(private_key);
+        let signer = Box::new(PrivateKeyEthereumSigner::new(pk_32)) as Box<dyn EcdsaSign + Send + Sync>;
+        signer
+    } else {
+        let signer = Box::new(SgxEthereumSigner::new(eid)) as Box<dyn EcdsaSign + Send + Sync>;
+        signer
+    }
+}
+
+fn get_signing_address(eid: sgx_enclave_id_t) -> Result<String, EnclaveError> {
+    Ok(equote::get_register_signing_address(eid).or(Err(EnclaveError::UnDetailedEnclaveErr))?.to_hex())
+}
+
+fn get_ethereum_address(eid: sgx_enclave_id_t, config: KMConfig) -> Result<String, EnclaveError> {
+    if config.with_private_key {
+        return Ok(config.account_address.clone());
+    }
+    Ok(equote::get_ethereum_address(eid).or(Err(EnclaveError::UnDetailedEnclaveErr))?.to_hex())
+}
+
+#[logfn(INFO)]
+pub fn start(eid: sgx_enclave_id_t) -> Result<(), ControllerError> {
+    let opt = controller::options::Opt::from_args();
+    let config = KMConfig::load_config(opt.principal_config.as_str())?;
+
+    let mut path = storage_dir(ENCLAVE_DIR).map_err(|_| ControllerError::VerifierError(VerifierError::CreateErr))?;
```

Comment:
```suggestion
    let mut path = storage_dir(ENCLAVE_DIR).or(ControllerError::VerifierError(VerifierError::CreateErr))?;
```

Reply:
There are a bunch of these... if you're doing a `.map_err(|_| x)` you probably want to `.or(x)` or `.or_else(|| x)`